src/rdkafka_transport.c

```diff
@@ -1403,6 +1403,15 @@ static void rd_kafka_transport_io_event (rd_kafka_transport_t *rktrans,
 
 	case RD_KAFKA_BROKER_STATE_AUTH:
 		/* SASL handshake */
+
+        if (events & POLLHUP) {
```

Comment:
We must read remaining socket buffer data if events&POLLIN, to make sure we receive any auth errors sent by the broker before it disconnects the socket.
So move this check down below the sasl_io_event() call.

Reply:
Done as requested.